lib/Interpreter/Interpreter.cpp

```diff
@@ -175,18 +190,29 @@ namespace cling {
     m_LookupHelper.reset(new LookupHelper(new Parser(PP, SemaRef,
                                                      /*SkipFunctionBodies*/false,
                                                      /*isTemp*/true), this));
+    if (!m_LookupHelper)
+      return;
 
-    if (!isInSyntaxOnlyMode())
+    if (!isInSyntaxOnlyMode()) {
       m_Executor.reset(new IncrementalExecutor(SemaRef.Diags,
                                                getCI()->getCodeGenOpts()));
+      if (!m_Executor)
+        return;
+    }
 
     // Tell the diagnostic client that we are entering file parsing mode.
     DiagnosticConsumer& DClient = getCI()->getDiagnosticClient();
     DClient.BeginSourceFile(getCI()->getLangOpts(), &PP);
 
     llvm::SmallVector<IncrementalParser::ParseResultTransaction, 2>
       IncrParserTransactions;
-    m_IncrParser->Initialize(IncrParserTransactions, isChildInterp);
+    if (!m_IncrParser->Initialize(IncrParserTransactions, isChildInterp)) {
```

Comment:
I am opposed to this approach in general. I think we should always have a well-constructed and initialized interpreter, this reduces a lot of debugging. What would be the use case here?